force_wfmanager/ui/setup/mco/mco_parameter_view.py

```diff
@@ -0,0 +1,312 @@
+
+    # -------------------
+    # Required Attributes
+    # -------------------
+
+    #: MCO parameter model
+    model = Instance(BaseMCOParameter)
+
+    # ------------------
+    # Dependent Attributes
+    # ------------------
+
+    #: Defines if the MCO parameter is valid or not. Updated by
+    #: :func:`verify_tree
+    #: <force_wfmanager.ui.setup.workflow_tree.WorkflowTree.verify_tree>`
+    valid = Bool(True)
+
+    #: An error message for issues in this modelview. Updated by
+    #: :func:`verify_tree
+    #: <force_wfmanager.ui.setup.workflow_tree.WorkflowTree.verify_tree>`
+    error_message = Unicode()
+
+    #: Event to request a verification check on the workflow
+    #: Listens to: :attr:`model.name <model>` and :attr:`model.type <model>`
+    verify_workflow_event = Event
+
+    # ----------
+    # Properties
+    # ----------
+
+    #: The human readable name of the MCO parameter class
+    label = Property(Unicode(), depends_on="model.[name,type]")
+
+    # ----------
+    #    View
+    # ----------
+
+    def default_traits_view(self):
+        """Default view containing both traits from the base class and
+        any additional user-defined traits"""
+
+        return View(Item('name', object='model'),
+                    Item('type', object='model'),
+                    Item('model',
+                         editor=InstanceEditor(),
+                         style='custom')
+                    )
+
+    #: Defaults
+    def _label_default(self):
+        """Return a default label corresponding to the MCO parameter factory"""
+        return get_factory_name(self.model.factory)
+
+    #: Property getters
+    @cached_property
+    def _get_label(self):
+        """Return a label appending both the parameter name and type to the
+        default"""
+        if self.model.name == '' and self.model.type == '':
+            return self._label_default()
+        return self._label_default()+': {type} {name}'.format(
+            type=self.model.type, name=self.model.name
+        )
+
+    #: Listeners
+    # Workflow Validation
+    @on_trait_change('model.[name,type]')
+    def parameter_change(self):
+        """Alert to a change in the model"""
+        self.verify_workflow_event = True
+
+
+class MCOParameterView(HasTraits):
+
+    # -------------------
+    # Required Attributes
+    # -------------------
+
+    #: MCO model
+    model = Instance(BaseMCOModel)
+
+    # -------------------
+    # Regular Attributes
+    # -------------------
+
+    #: List of MCO parameter model views
+    parameter_model_views = List(Instance(MCOParameterModelView))
+
+    # --------------------
+    # Dependent Attributes
+    # --------------------
+
+    #: The selected parameter model view
+    selected_parameter = Instance(MCOParameterModelView)
+
+    #: Creates new instances of MCO Parameters
+    parameter_entity_creator = Instance(NewEntityCreator)
+
+    #: Defines if the MCO parameter is valid or not. Updated by
```

Comment:
Do we need `valid`, `error_message` and `verify_workflow_event` if this class is a grouping of `MCOParameterModelViews`? Hopefully the verification from the parameters themselves should be enough, and if they're okay then the view will be as well.

Reply:
The traits have been left in at the moment to replace any verification checks referring to a group of KPIs or parameters that are missing in the BDSS (this is more obvious in the `KPISpecificationView`) For instance, do any KPI or parameter share the same name?

Ultimately it would be great to remove these checks from the UI but it feels that this requires some further consideration (hopefully by addressing force-h2020/force-bdss#213